test/Test/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs

```diff
@@ -251,10 +286,36 @@ instance FromModel (Int -> Int) where
   fromModel (FnNotInWHNF f) k = fromModel f $ \f' -> k (if ack 3 3 > 0 then f' else f')
```

Comment:
You can think of this as `\x -> notStrict b (ack 5 n) x`, and so is definitely in WHNF (but since contains thunks inside, not in NF).